src/Process.ts

```diff
@@ -75,10 +77,33 @@ export class Process {
         try {
             eventStream = new IonEventStream(ionReader, path, args);
 
-            // processes input stream into event stream
-            eventStream.writeEventStream(ionOutputWriter);
+            ionOutputWriter.writeSymbol(this.EVENT_STREAM);
+            let writeEvents: IonEvent[] = [];
+            for(let i =0; i < eventStream.getEvents().length; i++) {
+                let event = eventStream.getEvents()[i];
+
+                //process event stream into event stream
+                if(eventStream.isEventStream) {
+                    writeEvents = eventStream.getEvents();
+                    break;
+                }
```

Comment:
It looks like this case could be lifted out above the `for` loop? Consider making a helper function for collecting input stream events into an array.

```ts
if (eventStream.isEventStream) {
   writeEvents = eventStream.getEvents();
} else {
   writeEvents = collectInputStreamEvents(...); // <-- new helper function
}

for(let event of writeEvents) {
   event.write(ionOutputWriter);
}
```

That will simplify this function's responsibilities and make it easier to understand.